experimental/unitsconversion/src/measureunit.rs

```diff
@@ -0,0 +1,243 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use zerotrie::ZeroTrie;
+use zerovec::ZeroVec;
+
+use crate::{
+    provider::{Base, MeasureUnitItem, SiPrefix},
+    ConversionError,
+};
+
+// TODO(#4369): split this struct to two structs: MeasureUnitParser for parsing the identifier and MeasureUnit to represent the unit.
+pub struct MeasureUnit {
+    /// Contains the processed units.
+    pub contained_units: Vec<MeasureUnitItem>,
+}
+
+impl MeasureUnit {
+    // TODO: consider returning Option<(u8, &str)> instead of (1, part) for the case when the power is not found.
+    // TODO: complete all the cases for the powers.
+    // TODO: consider using a trie for the powers.
+    /// Get the power of the unit.
+    /// NOTE:
+    ///    if the power is found, the function will return (power, part without the power).
+    ///    if the power is not found, the function will return (1, part).
+    fn get_power(part: &str) -> Option<i8> {
```

Comment:
Nit (optional): I prefer free functions instead of static functions in Rust except for constructors.

Reply:
I was thinking about this too. I have moved them to si+prefix.rs and power.rs